From 0850abcd9558248f251ee8f676561915b2287cc6 Mon Sep 17 00:00:00 2001
From: Olivier Monnier <olivier.monnier@intel.com>
Date: Wed, 13 Mar 2013 10:25:03 +0100
Subject: [PORT FROM MAIN] Create non ODEX jar target for external releases

BZ: 91586

For external releases of java libraries, we must deliver
the non odex version

Change-Id: I4d0e4dd6d1c5a37fbc5757c87118d447faa151d3
Orig-Change-Id: I0dbb8988d05ec9a8670f181c5a2662cbec0ed875
Signed-off-by: Olivier Monnier <olivier.monnier@intel.com>
Reviewed-on: http://android.intel.com:8080/96333
Reviewed-by: Alberdi, Ion <ion.alberdi@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 core/java_library.mk |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/core/java_library.mk b/core/java_library.mk
index 67c1bcb..2dbe93f 100644
--- a/core/java_library.mk
+++ b/core/java_library.mk
@@ -119,10 +119,15 @@ $(built_odex) : $(common_javalib.jar) | $(DEXPREOPT) $(DEXOPT)
 
 $(LOCAL_BUILT_MODULE) : $(common_javalib.jar) | $(ACP)
 	$(call copy-file-to-target)
+	$(hide) $(ACP) $@ $(patsubst %.jar,%.jar.dex,$@)
 ifneq (nostripping,$(LOCAL_DEX_PREOPT))
 	$(call dexpreopt-remove-classes.dex,$@)
 endif
 
+# non odex jar is saved .jar.dex for external releases
+built_dexjar := $(basename $(LOCAL_BUILT_MODULE)).jar.dex
+$(built_dexjar): $(LOCAL_BUILT_MODULE)
+
 endif # dexpreopt_boot_jar_module
 
 else # LOCAL_DEX_PREOPT
-- 
1.7.4.1

